### PR TITLE
Removing Hilbert and Preparing for Graph

### LINKED ISSUE #372: Remove C++ hilbert classes

### PR DESCRIPTION
**Pull request by gcarleo**
Removes most of the remaining C++ code. Only Lattice and MPS to be merged later. 

closes #372

### CODE REVIEW

**Comment by gcarleo** [comment]
OK thanks @VolodyaCO I wasn't used to passing also the list of nodes, but indeed it makes sense. I have fixed the tests now, we will merge this before the Lattice PR, so that you can update your lattice PR to reflect the changes I did to the Graph interface in this one

**Comment by gcarleo** [comment]
Merging this since it's mostly removing stuff

File: netket/graph/abstract_graph.py [hunk 1]
```diff
@@ -1,17 +1,14 @@
...
 class AbstractGraph(abc.ABC):
     """Abstract class for NetKet graph objects"""
 
-    @property
     @abc.abstractmethod
     def is_connected(self):
         r"""bool: True if the graph is connected"""
         raise NotImplementedError
 
-    @property
```
Comment (PhilipVinc) [inline netket/graph/abstract_graph.py:14]: why are those no longer properties?

Comment (gcarleo) [reply netket/graph/abstract_graph.py:14]: mostly for consistency with what happens internally in networkx: basically knowing if it is connected involves doing some operations, it's not a static attribute. Same for automorphisms etc

File: netket/graph/graph.py [hunk 2]
```diff
@@ -163,8 +160,22 @@ def Edgeless(nodes):
...
+        if not isinstance(nodes, int):
+            raise TypeError("nodes must be either an integer or a list")
+        nodes = range(nodes)
 
     edgelessgraph = _nx.MultiGraph()
     edgelessgraph.add_nodes_from(nodes)
     return NetworkX(edgelessgraph)
+
+
+def DoubledGraph(graph):
```
Comment (PhilipVinc) [inline netket/graph/graph.py:172]: more a reminder for me than anything: this doubled graph does not correctly set automorphisms...

**Pull request merged**